internal/catalog/internal/types/service_endpoints_test.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	defaultEndpointTenancy = &pbresource.Tenancy{
+		Partition: "default",
+		Namespace: "default",
+		PeerName:  "local",
+	}
+)
+
+func createServiceEndpointsResource(t *testing.T, data protoreflect.ProtoMessage) *pbresource.Resource {
+	res := &pbresource.Resource{
+		Id: &pbresource.ID{
+			Type:    ServiceEndpointsType,
+			Tenancy: defaultEndpointTenancy,
+			Name:    "test-service",
+		},
+	}
+
+	var err error
+	res.Data, err = anypb.New(data)
+	require.NoError(t, err)
+	return res
+}
+
+func TestValidateServiceEndpoints_Ok(t *testing.T) {
+	data := &pbcatalog.ServiceEndpoints{
+		Endpoints: []*pbcatalog.Endpoint{
+			{
+				TargetRef: &pbresource.ID{
+					Type:    WorkloadType,
+					Tenancy: defaultEndpointTenancy,
+					Name:    "foo",
+				},
+				Addresses: []*pbcatalog.WorkloadAddress{
+					{
+						Host: "198.18.0.1",
+					},
+				},
+				Ports: map[string]*pbcatalog.WorkloadPort{
+					"foo": {
+						Port:     8443,
+						Protocol: pbcatalog.Protocol_PROTOCOL_MESH,
+					},
+				},
+				HealthStatus: pbcatalog.Health_HEALTH_PASSING,
+			},
+		},
+	}
+
+	res := createServiceEndpointsResource(t, data)
+
+	err := ValidateServiceEndpoints(res)
+	require.NoError(t, err)
+}
+
+func TestValidateServiceEndpoints_ParseError(t *testing.T) {
+	// Any type other than the ServiceEndpoints type would work
+	// to cause the error we are expecting
+	data := &pbcatalog.IP{Address: "198.18.0.1"}
+
+	res := createServiceEndpointsResource(t, data)
+
+	err := ValidateServiceEndpoints(res)
+	require.Error(t, err)
+	require.True(t, errors.As(err, &resource.ErrDataParse{}))
+}
+
+func TestValidateServiceEndpoints_EndpointInvalid(t *testing.T) {
+	genData := func() *pbcatalog.Endpoint {
+		return &pbcatalog.Endpoint{
+			TargetRef: &pbresource.ID{
+				Type:    WorkloadType,
+				Tenancy: defaultEndpointTenancy,
+				Name:    "foo",
+			},
+			Addresses: []*pbcatalog.WorkloadAddress{
+				{
+					Host:  "198.18.0.1",
+					Ports: []string{"foo"},
+				},
+			},
+			Ports: map[string]*pbcatalog.WorkloadPort{
+				"foo": {
+					Port: 42,
+				},
+			},
+			HealthStatus: pbcatalog.Health_HEALTH_PASSING,
+		}
+	}
+
+	type testCase struct {
+		modify      func(*pbcatalog.Endpoint)
+		validateErr func(t *testing.T, err error)
+	}
+
+	cases := map[string]testCase{
+		"invalid-target": {
+			modify: func(endpoint *pbcatalog.Endpoint) {
+				endpoint.TargetRef.Type = NodeType
+			},
+			validateErr: func(t *testing.T, err error) {
+				require.True(t, errors.Is(err, resource.ErrInvalidReferenceType{AllowedType: WorkloadType}))
+			},
+		},
+		"invalid-address": {
+			modify: func(endpoint *pbcatalog.Endpoint) {
+				endpoint.Addresses[0].Ports = []string{"bar"}
+			},
+			validateErr: func(t *testing.T, err error) {
+				require.True(t, errors.Is(err, errInvalidPortReference{Name: "bar"}))
+			},
+		},
```

Comment:
Can you use `require.ErrorIs(...)` and `require.ErrorAs(...)` instead of `require.True(...)` in all the test cases?

Reply:
ooh, I didn't know those existed, we absolutely should use those instead.